lib/node_modules/@stdlib/stats/base/ndarray/nanmax-by/docs/types/index.d.ts

```diff
@@ -0,0 +1,100 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { typedndarray } from '@stdlib/types/ndarray';
+
+/**
+* Callback function applied to each element of an ndarray.
+*
+* @returns result
+*/
+type Nullary<ThisArg> = ( this: ThisArg ) => number | void;
+
+/**
+* Callback function applied to each element of an ndarray.
+*
+* @param value - current array element
+* @returns result
+*/
+type Unary<T, ThisArg> = ( this: ThisArg, value: T ) => number | void;
+
+/**
+* Callback function applied to each element of an ndarray.
+*
+* @param value - current array element
+* @param index - current array element index
+* @returns result
+*/
+type Binary<T, ThisArg> = ( this: ThisArg, value: T, index: number ) => number | void;
+
+/**
+* Callback function applied to each element of an ndarray.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param array - input ndarray
+* @returns result
+*/
+type Ternary<T, U, ThisArg> = ( this: ThisArg, value: T, index: number, array: U ) => number | void;
+
+/**
+* Callback function applied to each element of an ndarray.
+*
+* @param value - current array element
+* @param index - current array element index
+* @param array - input ndarray
+* @returns result
+*/
+type Callback<T, U, ThisArg> = Nullary<ThisArg> | Unary<T, ThisArg> | Binary<T, ThisArg> | Ternary<T, U, ThisArg>;
+
+
+/**
+* Computes the maximum value of a one-dimensional ndarray via a callback function while ignoring `NaN` values.
+*
+* If provided an empty ndarray or one containing only `NaN` values, the function returns `NaN`.
```

Comment:
This isn't how we do notes. See https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/stats/strided/nanmax-by/docs/types/index.d.ts#L130.